### PR TITLE
Iframe api refactor v2

### PR DESCRIPTION
**Pull request by moufmouf**
This PR builds upon #1085.

It changes the way the WA object is constructed (to something a lot more simple... but a little bit less magic).

@jonnytest1 I did love your type system wizardry to recreate the WA types from several classes and adding back the deprecated methods automatically but:

- we could not easily write the list of methods of a class that should be backported to root (so the methods of a Command Class has to be all ported to root or not)
- by writing the list of deprecated methods manually, I can tag them with @deprecated and add messages in console.log

@jonnytest1 I kept the registeredCallback mechanism which is super dope, and reorganized / renamed the methods in different classes.

Tell me if you see any things to change ;)

### CODE REVIEW

**Comment by jonnytest1** [comment]
🤔 i think at least console.warn would be appropriate 
we could also (sneakily add a removal date already to create some kind of pressure to not use them :3 wether we actually remember it is a different matter ^^ )

File: front/src/Api/iframe/ui.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                })
+            }
+        });
+
+        popups.set(popupId, popup)
+        return popup;
+    }
+
+    displayBubble(): void {
+        window.parent.postMessage({ 'type': 'displayBubble' }, '*');
```
Comment (jonnytest1) [inline front/src/Api/iframe/ui.ts:106]: oh this is still using the untyped and raw way to post to the engine

**Comment by jonnytest1** [comment]
registeredCallbacks.ts  is an empty file

**Comment by jonnytest1** [comment]
movin the apiCallback method to registered_callbacks probably makes more sense

**Comment by jonnytest1** [comment]
🤔 i dont think im a fan of destructuring imports 
those classes should be their own files anyways

**Comment by moufmouf** [comment]
> :thinking: i dont think im a fan of combining default and destructuring imports

I'm not sure I understand what you mean here.

Otherwise, I took your comments in account. Thanks for the feedback!

File: front/src/iframe_api.ts [hunk 2]
```diff
@@ -1,286 +1,173 @@
...
+    IframeResponseEventMap,
+    isIframeResponseEventWrapper,
+    TypedMessageEvent
+} from "./Api/Events/IframeEvent";
+import chat from "./Api/iframe/chat";
+import type {IframeCallback} from './Api/iframe/IframeApiContribution';
+import nav from "./Api/iframe/nav";
+import controls from "./Api/iframe/controls";
+import ui, {ButtonDescriptor, Popup} from "./Api/iframe/ui";
+import sound, {Sound} from "./Api/iframe/sound";
```
Comment (jonnytest1) [inline front/src/iframe_api.ts:13]: these imports seems weird - and should also not be in the same file it just gets unreadable

Comment (moufmouf) [reply front/src/iframe_api.ts:13]: Indeed! :+1:

**Review by jonnytest1 — APPROVED** [review]

**Pull request merged**